apps/main/src/llamalend/widgets/borrow/llama.util.ts

```diff
@@ -1,17 +1,27 @@
-import type { LendMarketTemplate } from '@curvefi/llamalend-api/lib/lendMarkets'
-import type { MintMarketTemplate } from '@curvefi/llamalend-api/lib/mintMarkets'
+import { ZeroAddress } from 'ethers'
+import type { SetValueConfig } from 'react-hook-form'
+import { LendMarketTemplate } from '@curvefi/llamalend-api/lib/lendMarkets'
 import { requireLib } from '@ui-kit/features/connect-wallet'
-import { LlamaMarketType } from '@ui-kit/types/market'
+import { BorrowPreset, type LlamaMarketTemplate } from './borrow.types'
 
-type MarketAndType =
-  | [market: MintMarketTemplate, type: LlamaMarketType.Mint]
-  | [market: LendMarketTemplate, type: LlamaMarketType.Lend]
-
-export function getLlamaMarket(id: string): MarketAndType {
+export function getLlamaMarket(id: string): LlamaMarketTemplate {
   const lib = requireLib('llamaApi')
   const mintMarket = lib.getMintMarket(id)
-  if (mintMarket) return [mintMarket, LlamaMarketType.Mint] as const
+  if (mintMarket) return mintMarket
   const lendMarket = lib.getLendMarket(id)
-  if (lendMarket) return [lendMarket, LlamaMarketType.Lend] as const
+  if (lendMarket) return lendMarket
```

Comment:
`return mintMarket || lendMarket || (() => { throw new Error(`Market with ID ${id} not found`) })()`

🤓 

Too bad you can't throw in a return statement 😔